src/fidesops/service/connectors/query_config.py

```diff
@@ -666,15 +664,47 @@ def generate_query(
         for string_path, reference_values in filtered_data.items():
             for value in reference_values:
                 request_params.append(
-                    self.prepare_params(current_request, {string_path: value})
+                    self.prepare_query_params(current_request, {string_path: value})
                 )
         logger.info(f"Populated request params for {current_request.path}")
         return request_params
 
+    @staticmethod
+    def prepare_update_params(
+        request: SaaSRequest, param_values: Dict[str, Any], body: Dict[str, Any]
+    ) -> SaaSRequestParams:
+        """
+        Populates the placeholders in the request with the given param values
+        """
+        path: str = request.path
+        params: Dict[str, Any] = {}
+
+        for param in request.request_params:
+            if param.type == "query":
+                if param.default_value:
+                    params[param.name] = param.default_value
+                elif param.references:
+                    params[param.name] = pydash.get(
+                        param_values, param.references[0].field
+                    )
+                elif param.identity:
+                    params[param.name] = pydash.get(param_values, param.identity)
+            elif param.type == "path":
+                path = path.replace(
+                    f"<{param.name}>",
+                    pydash.get(param_values, param.references[0].field),
+                )
```

Comment:
There is a lot of duplicated logic from `prepare_query_params` - it seems like prepare_query_params could also use the pydash methods you're using here.  Is there any way to consolidate?

I may not fully understand the extent of the differences that made you feel you needed to separate the two.

Reply:
The `param_values` param in `prepare_query_params` is a map of request_param.names to values. So if a request_param is defined as

```
request_params:
  - name: query
    type: query
    identity: email
    data_type: string
```
we can just use the name value `query` as the dictionary key

However the `param_values` param in `prepare_update_params` is a map of fields to values. So for something like this
```
- name: subscriber_hash
  type: path
  references:
    - dataset: mailchimp_connector_example
      field: member.id
      direction: from
```
We have to use the `field` value as the key instead of name (`subscriber_hash` in this case)

Reply:
thanks for the clarification!  The type signatures being the same makes this confusing, is the param_values type for `prepare_update_params` possibly `Dict[str, Row]`? I might make variable names more distinct for `prepare_update_params` and add more detail to the docstring.